### PR TITLE
array.d: add debug stomping code

### PR DESCRIPTION
**Pull request by WalterBright**
to flush out any dangling pointers

### CODE REVIEW

File: src/dmd/root/array.d [hunk 1]
```diff
@@ -137,12 +143,30 @@ public:
...
+            debug (stomp)
+            {
+                if (length < data.length)
+                    memset(data.ptr + length, 0xFF, (data.length - length) * T.sizeof);
+            }
+            else
+            {
+                if (mem.isGCEnabled)
+                    if (length < data.length)
+                        memset(data.ptr + length, 0xFF, (data.length - length) * T.sizeof);
```
Comment (rainers) [inline src/dmd/root/array.d]: There is a small change for the non-stomping case here: the existing implementation assumes that the reserved space will be overwritten by the caller anyway, so it would not have to be reset here. It usually won't make a difference but for inserting large arrays, but is this change deliberate?

Comment (RazvanN7) [reply src/dmd/root/array.d]: ping @WalterBright

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/root/array.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/root/array.d]: Yes. What if the caller doesn't overwrite it? The stomping code doesn't need to be efficient, as it is only enabled for debugging porpoises.

Comment (rainers) [reply src/dmd/root/array.d]: > The stomping code doesn't need to be efficient, as it is only enabled for debugging porpoises.

This is not debug code, but code executed if the GC is enabled.
